wordpress-mobile/WordPress-iOS
# [JP Plugin] Make the current Jetpack Install view configurable

Refs #20018
Depends on #20063

This PR makes the `JetpackRemoteInstallStateView` configurable, so the buttons and strings can be set differently when it's reused for the full Jetpack installation process.

Previously, the view class directly reads the `JetpackRemoteInstallState` object and reads the strings from there. To make this configurable, I've abstracted this part into a delegate protocol instead, and the `JetpackRemoteInstallViewController` will be the one responsible for providing the view with the strings and controlling the visibility state of the views.

## To test

Verify that the existing Jetpack Remote Install flow is correct and not impacted by this change.

- Create a self-hosted site, and have the Jetpack plugin deactivated or uninstalled.
- Log into the site through the site address method (username & password).
- Navigate to the Notifications tab.
- Tap "Install Jetpack", and verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Install Jetpack"`
  - 🔎 The description text is correct: `"Your website credentials will not be stored and are used only for the purpose of installing Jetpack."`
  - 🔎 The primary button, "Continue", is displayed.
- To test the error screen, set your device to airplane mode and tap "Continue".
- Verify that an error screen is shown, and:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"There was a problem"`
  - 🔎 The description text is correct: `"Jetpack could not be installed at this time."`
  - 🔎 The primary button, "Retry", is shown.
  - 🔎 The Contact Support button is shown.
- Turn off your airplane mode and tap "Retry".
- This should briefly show you a loading screen. Verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Installing Jetpack"`
  - 🔎 The description text is correct: `"Installing Jetpack on your site. This can take up to a few minutes to complete."`
  - 🔎 The primary button is hidden.
  - 🔎 A loading indicator is shown.
- After the loading completes and the complete screen is shown, verify that:
  - 🔎 The correct image is displayed.
  - 🔎 The title text is correct: `"Jetpack installed"`
  - 🔎 The description text is correct: `"Now that Jetpack is installed, we just need to get you set up. This will only take a minute."`
  - 🔎 The primary button, "Set up", is shown.

## Regression Notes
1. Potential unintended areas of impact
Might impact the strings of the current Jetpack Remote Install flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:
